File: src/shopperior-ng/src/app/core/data/shopperior-api/models/shopping-list-dto.ts
import { CategoryDto } from "./category-dto";
import { ListItemDto } from "./list-item-dto";
import { UserListPermissionDto } from "./user-list-permission-dto";

export interface ShoppingListDto {
  guid: string;
  name: string;
  permissions: UserListPermissionDto[];
  categories: CategoryDto[];
  items: ListItemDto[];
}

File: src/Shopperior.Api/Shopperior.WebApi/ShoppingLists/Models/ShoppingListDto.cs
namespace Shopperior.WebApi.ShoppingLists.Models;

public class ShoppingListDto
{
    public Guid Guid { get; set; }
    public string Name { get; set; }
    public IEnumerable<UserListPermissionDto> Permissions { get; set; }
    public IEnumerable<ListItemDto> Items { get; set; }
}